File: DI/DI-Dagger-Java/src/main/java/model/persona/services/PersonaFileStorage.java
package model.persona.services;

import model.persona.model.Persona;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.UUID;

@Singleton
public class PersonaFileStorage implements IPersonaStorage {
    private final String id = UUID.randomUUID().toString();

    @Inject
    public PersonaFileStorage() {

    }

    @Override
    public Persona save(Persona item) {
        System.out.println("PersonaDataBaseStorage.save() --> " + item);
        return item;
    }

    @Override
    public String toString() {
        return "PersonaFileStorage{" +
                "id='" + id + '\'' +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/personas/services/PersonaFileStorage.kt
package models.personas.services

import models.personas.model.Persona
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton


@Singleton
class PersonaFileStorage
@Inject constructor() : IPersonaStorage {
    private val id = UUID.randomUUID().toString()
    override fun save(item: Persona): Persona {
        println("PersonaFileStorage.save() --> $item")
        return item
    }

    override fun toString() = "PersonaFileStorage(id='$id')"
}
